students/TimPauley/session04/test_customer.py

```diff
@@ -0,0 +1,238 @@
+    customer_dict = search_customer("W3434ff")
+    second_customer["credit_limit"] = float(second_customer["credit_limit"])
+    assert customer_dict == second_customer
+    clean_up(ok_customer["customer_id"])
+    clean_up(second_customer["customer_id"])
+
+
+'''
+Rename fuctional test
+'''
+def test_search_customer_null():
+    add_customer(**ok_customer)
+    customer_dict = search_customer("12345")
+    assert customer_dict == {}
+    customer_dict = search_customer(7)
+    assert customer_dict == {}
+    customer_dict = search_customer()
+    assert customer_dict == {}
+    clean_up(ok_customer["customer_id"])
+
+'''
+Rename fuctional test
+'''
+def test_delete_customer():
+    add_customer(**ok_customer)
+    second_customer = add_second_customer()
+    delete_customer(second_customer["customer_id"])
+    customer_dict = search_customer(second_customer["customer_id"])
+    assert customer_dict == {}
+    ok_customer["credit_limit"] = float(ok_customer["credit_limit"])
+    customer_dict = search_customer(ok_customer["customer_id"])
+    assert customer_dict == ok_customer
+    clean_up(ok_customer['customer_id'])
+
+'''
+Rename fuctional test
+'''
+def test_delete_not_a_customer():
+    add_customer(**ok_customer)
+    second_customer = add_second_customer()
+    delete_customer("12345")
+    delete_customer(7)
+    delete_customer()
+    clean_up(ok_customer['customer_id'])
+    clean_up(second_customer['customer_id'])
+
+'''
+Rename fuctional test
+'''
+def test_update_customer_credit():
+    add_customer(**ok_customer)
+    second_customer = add_second_customer()
+    update_customer(
+        customer_id=second_customer["customer_id"], credit_limit=100.0)
+    customer_dict = search_customer(second_customer["customer_id"])
+    assert customer_dict["credit_limit"] == 100.0
+    clean_up(ok_customer['customer_id'])
+    clean_up(second_customer['customer_id'])
+
+'''
+Rename fuctional test
+'''
+def test_update_customer_credit_no_match():
+    add_customer(**ok_customer)
+    second_customer = add_second_customer()
+    with pytest.raises(ValueError):
+        update_customer(customer_id="12345", credit_limit=100.0)
+    clean_up(ok_customer['customer_id'])
+    clean_up(second_customer['customer_id'])
+
+'''
+Rename fuctional test
+'''
+def test_list_active_customers():
+    add_customer(**ok_customer)
+    second_customer = add_second_customer()
+    third_customer = add_third_customer()
+    fourth_customer = add_fourth_customer()
+    active_customers = list_active_customers()
+    assert active_customers == 2
+    test_customer3 = Customer.get(
+        Customer.customer_id == third_customer["customer_id"])
+    test_customer3.status = True
+    test_customer3.save()
+    active_customers = list_active_customers()
+    assert active_customers == 3
+    clean_up(ok_customer['customer_id'])
+    clean_up(second_customer['customer_id'])
+    clean_up(third_customer['customer_id'])
+    clean_up(fourth_customer['customer_id'])
```

Comment:
A lot of good tests in here. Not all of the tests that say rename functional test should really be functional. Some of the tests seem like unit tests with different inputs, and would be better to use parameterize and keep a unit test.